ocaml/typing/jkind.ml

```diff
@@ -404,6 +404,22 @@ module Const = struct
         name = "immutable_data"
       }
 
+    let mutable_data =
+      { jkind =
+          { layout = Sort Value;
+            modes_upper_bounds =
+              { linearity = Linearity.Const.min;
+                contention = Contention.Const.max;
+                portability = Portability.Const.min;
+                uniqueness = Uniqueness.Const.max;
+                areality = Locality.Const.max
+              };
+            externality_upper_bound = Externality.max;
+            nullability_upper_bound = Nullability.Non_null
+          };
+        name = "mutable_data"
```

Comment:
Check discussion on Slack about the name here.

Reply:
I will rename `immutable_data` to `data` in a separate PR. `mutable_data` stays as it is.